app/electron/main.ts

```diff
@@ -415,6 +415,49 @@ function startElecron() {
       mainWindow = null;
     });
 
+    // Force Single Instance Application
+    const gotTheLock = app.requestSingleInstanceLock();
+    if (gotTheLock) {
+      app.on('second-instance', () => {
+        // Someone tried to run a second instance, we should focus our window.
+        if (mainWindow) {
+          if (mainWindow.isMinimized()) mainWindow.restore();
+          mainWindow.focus();
+        }
+      });
+    } else {
+      app.quit();
+      return;
+    }
+
+    /*
+    if a library is trying to open a url other than app url in electron take it 
+    to the default browser
+    */
+    mainWindow.webContents.on('will-navigate', (event, url) => {
+      if (url.startsWith(startUrl)) {
+        return;
+      }
+      shell.openExternal(url);
+    });
+
+    app.on('open-url', (event, url) => {
+      mainWindow?.focus();
+      try {
+        const urlObj = new URL(url);
+        const urlParam = urlObj.hostname;
+        let massagedStartUrl = startUrl;
+        // this check helps us to avoid adding multiple / to the startUrl when appending the incoming url to it
+        if (massagedStartUrl.endsWith('/')) {
+          massagedStartUrl = massagedStartUrl.slice(0, startUrl.length - 1);
+        }
+        // load the index.html from build and route to the hostname received in the protocol handler url
+        mainWindow?.loadURL(massagedStartUrl + '#' + urlParam + urlObj.search);
```

Comment:
We should just include the code we may expect to throw exceptions. 
So that would just be the urlObj creation.

Reply:
@joaquimrocha there are other pieces in this code segment that are dependent on urlObj so I think they also will not work if URL is invalid.